zerver/lib/users.py

```diff
@@ -476,6 +486,125 @@ def format_user_row(
     return result
 
 
+def user_access_restricted_in_realm(target_user: UserProfile) -> bool:
+    if target_user.is_bot:
+        return False
+
+    realm = target_user.realm
+    if realm.can_access_all_users_group.name == SystemGroups.EVERYONE:
+        return False
+
+    return True
+
+
+def check_user_can_access_all_users(acting_user: Optional[UserProfile]) -> bool:
+    if acting_user is None:
+        return True
+
+    if not acting_user.is_guest:
+        return True
+
+    realm = acting_user.realm
+    if is_user_in_group(realm.can_access_all_users_group, acting_user):
```

Comment:
We can directly check whether the group is set to "Everyone" here because we only provide two options here currently.